<?xml version="1.0"?>
<robot name="simple_sphere">
    <material name="blue">
        <color rgba="0.2 0.2 1 1"/>
    </material>
    <link name="sphere_link">
        <!-- Inertial properties -->
        <inertial>
            <mass value="0.1"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="0.0002" ixy="0.0" ixz="0.0" iyy="0.0002" iyz="0.0" izz="0.0002"/>
        </inertial>

        <!-- Collision shape -->
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <sphere radius="0.025"/>
            </geometry>
        </collision>

        <!-- Visual shape with red material -->
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
              <sphere radius="0.025"/>
            </geometry>
            <material name="blue">
              <color rgba="0.2 0.2 1 1"/>
            </material>
          </visual>


    </link>


    <gazebo reference="sphere_link">
        <material>Gazebo/Orange</material>
    </gazebo>
    <gazebo>
    <plugin name="libgazebo_ros_p3d" filename="libgazebo_ros_p3d.so">
        <ros>
            <namespace>ball</namespace>
            <remapping>odom:=odom</remapping>
            </ros>
            <frame_name>world</frame_name>
            <body_name>sphere_link</body_name>
            <update_rate>200.0</update_rate>
            <gaussian_noise>0.01</gaussian_noise>
            
        </plugin>
    </gazebo>

</robot>
